core/test/scripts/Voting.js

```diff
@@ -236,48 +297,116 @@ contract("scripts/Voting.js", function(accounts) {
   });
 
   it("Constant price", async function() {
-    const identifier = web3.utils.utf8ToHex("Custom Index (1)");
-    const time = "1560762000";
+    const identifier = TEST_IDENTIFIERS["Custom Index (1)"].key;
+    const time = await voting.getCurrentTime();
 
     // Request an Oracle price.
-    await supportedIdentifiers.addSupportedIdentifier(identifier);
     await voting.requestPrice(identifier, time);
 
     const votingSystem = new VotingScript.VotingSystem(voting, voter, [new MockNotifier()]);
     await moveToNextRound(voting);
-    await votingSystem.runIteration();
+    let result = await votingSystem.runIteration();
+    assert.equal(result.batches, 1);
+    assert.equal(result.updates.length, 1);
+    assert.equal(result.skipped.length, 0);
+    assert.equal(result.failures.length, 0);
     await moveToNextPhase(voting);
-    await votingSystem.runIteration();
+    result = await votingSystem.runIteration();
+    assert.equal(result.batches, 1);
+    assert.equal(result.updates.length, 1);
     await moveToNextRound(voting);
 
-    assert.equal((await voting.getPrice(identifier, time)).toString(), web3.utils.toWei("1"));
+    assert.equal(
+      (await voting.getPrice(identifier, time)).toString(),
+      TEST_IDENTIFIERS["Custom Index (1)"].expectedPrice
+    );
   });
 
   it("Numerator/Denominator", async function() {
-    const identifier = web3.utils.utf8ToHex("0.5");
-    const time = "1560762000";
-
-    VotingScript.SUPPORTED_IDENTIFIERS["0.5"] = {
-      numerator: {
-        dataSource: "Constant",
-        value: "1"
-      },
-      denominator: {
-        dataSource: "Constant",
-        value: "2"
-      }
-    };
+    const identifier = TEST_IDENTIFIERS["0.5"].key;
+    const time = await voting.getCurrentTime();
+
     // Request an Oracle price.
-    await supportedIdentifiers.addSupportedIdentifier(identifier);
     await voting.requestPrice(identifier, time);
 
     const votingSystem = new VotingScript.VotingSystem(voting, voter, [new MockNotifier()]);
     await moveToNextRound(voting);
-    await votingSystem.runIteration();
+    let result = await votingSystem.runIteration();
+    assert.equal(result.batches, 1);
+    assert.equal(result.updates.length, 1);
+    assert.equal(result.skipped.length, 0);
+    assert.equal(result.failures.length, 0);
     await moveToNextPhase(voting);
-    await votingSystem.runIteration();
+    result = await votingSystem.runIteration();
+    assert.equal(result.batches, 1);
+    assert.equal(result.updates.length, 1);
+    await moveToNextRound(voting);
+
+    assert.equal((await voting.getPrice(identifier, time)).toString(), TEST_IDENTIFIERS["0.5"].expectedPrice);
+  });
+
+  it("Only batches up to the maximum number of commits or reveals that can fit in one block", async function() {
+    const identifier = TEST_IDENTIFIERS["Custom Index (100)"].key;
+    let testTransactions = Math.round(VotingScript.BATCH_MAX_COMMITS * 2 + 1);
```

Comment:
Is `Math.round()` necessary here?

Reply:
Hm I was using this because at one point I thought the `*` was converting an Int to a Float but this seems to work. I forget the context so I'm removing `Math.round()`